testutils/gke/client.go

```diff
@@ -109,12 +110,23 @@ func (gsc *sdkClient) GetCluster(project, region, zone, clusterName string) (*co
 	return gsc.Projects.Locations.Clusters.Get(clusterFullPath).Context(context.Background()).Do()
 }
 
+// ListClustersInProject lists all the GKE clusters created in the given project.
+func (gsc *sdkClient) ListClustersInProject(project string) ([]*container.Cluster, error) {
+	var clusters []*container.Cluster
```

Comment:
should this follow the pattern of
```
if zonal:
  list eveything in zone
else:
  list everything in project
```
?

Reply:
or maybe we can list both

Reply:
One difference between this function and others is `region` is not required. 

We can define the function as
```
func ListClusters(project, region, zone string) ([]*container.Cluster, error) {
  if zone != "" {
    return ...
  }
  
  if region != "" {
    return ...
  }

  return ...
}
```
But then if we want to get all clusters for this project, we'll have to pass two empty strings.

If you think this is better I can make the change.